playwright/main.py

```diff
@@ -59,9 +60,13 @@ async def run_driver_async() -> Connection:
 
 
 def run_driver() -> Connection:
-    loop = asyncio.get_event_loop()
-    if loop.is_running():
-        raise Error("Can only run one Playwright at a time.")
+    # if we're running in a thread, get_event_loop won't create an event loop for us, we'll need to make it ourselves
+    if not isinstance(threading.current_thread(), threading._MainThread):  # type: ignore
```

Comment:
We should probably reuse existing loop, regardless of the thread. I have a code health patch that resolves it as a driver-by: https://github.com/microsoft/playwright-python/pull/354/files

Reply:
I wasn't aware. Feel free to open if this is still relevant.

Reply:
This is basically checking for one of the conditions which will cause `RuntimeError` to be raised an avoiding it prematurely.